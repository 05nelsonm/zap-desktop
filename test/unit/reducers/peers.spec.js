import peersReducer, { GET_PEERS, RECEIVE_PEERS } from 'reducers/peers'

describe('reducers', () => {
  describe('peersReducer', () => {
    it('should handle initial state', () => {
      expect(peersReducer(undefined, {})).toMatchSnapshot()
    })

    it('should have GET_PEERS', () => {
      expect(GET_PEERS).toEqual('GET_PEERS')
    })

    it('should have RECEIVE_PEERS', () => {
      expect(RECEIVE_PEERS).toEqual('RECEIVE_PEERS')
    })

    it('should correctly getPeers', () => {
      expect(peersReducer(undefined, { type: GET_PEERS })).toMatchSnapshot()
    })

    it('should correctly receivePeers', () => {
      expect(peersReducer(undefined, { type: RECEIVE_PEERS, peers: 'foo' })).toMatchSnapshot()
    })
  })
})
