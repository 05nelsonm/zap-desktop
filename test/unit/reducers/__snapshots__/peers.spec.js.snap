// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers peersReducer should correctly getPeers 1`] = `
Object {
  "connecting": false,
  "disconnecting": false,
  "peers": Array [],
  "peersLoading": true,
}
`;

exports[`reducers peersReducer should correctly receivePeers 1`] = `
Object {
  "connecting": false,
  "disconnecting": false,
  "peers": "foo",
  "peersLoading": false,
}
`;

exports[`reducers peersReducer should handle initial state 1`] = `
Object {
  "connecting": false,
  "disconnecting": false,
  "peers": Array [],
  "peersLoading": false,
}
`;
